include/onnxruntime/core/framework/tensor.h

```diff
@@ -182,6 +183,13 @@ class Tensor final {
     return ret;
   }
 
+  /**
+  Replace the tensor's buffer with a ONNXRuntime C-Tenosr. The original buffer will be released!!!
+    p:  the replacement tensor
+	allocator:  the allocator for replacement buffer
+  */
+  Status ReplaceBuffer(ONNXRunTimeTensor* p, AllocatorPtr allocator, int64_t offset = 0);
```

Comment:
Aren't you making things more complex?

Reply:
yes I don't like it too. Like I mentioned in description, I prefer to leverage Ryan's custom op c api which has a c interface of op kernel context. This PR is a temp workaround since I don't want to break nGraph integration. 
It depends on the schedule, if we prefer to unify the custom op c api with func api now, I would abandon this pr.

Reply:
After discussion we prefer to solve it with custom op c api, so abandon this temp solution